diff --git a/Makefile b/Makefile
index b3cfd71..ea45121 100644
--- a/Makefile
+++ b/Makefile
@@ -13,6 +13,7 @@ ENABLE_TCL := 1
 ENABLE_ABC := 1
 ENABLE_GLOB := 1
 ENABLE_PLUGINS := 1
+ENABLE_GHDL := 0
 ENABLE_READLINE := 1
 ENABLE_EDITLINE := 0
 ENABLE_VERIFIC := 0
@@ -487,6 +489,14 @@ ifeq ($(ENABLE_PROTOBUF),1)
 LDLIBS += $(shell pkg-config --cflags --libs protobuf)
 endif
 
+ifeq ($(ENABLE_GHDL),1)
+GHDL_DIR ?= /usr/local/ghdl
+GHDL_INCLUDE_DIR ?= $(GHDL_DIR)/include
+GHDL_LIB_DIR ?= $(GHDL_DIR)/lib
+CXXFLAGS += -I$(GHDL_INCLUDE_DIR) -DYOSYS_ENABLE_GHDL
+LDLIBS += $(GHDL_LIB_DIR)/libghdl.a $(shell cat $(GHDL_LIB_DIR)/libghdl.link)
+endif
+
 ifeq ($(ENABLE_COVER),1)
 CXXFLAGS += -DYOSYS_ENABLE_COVER
 endif
